src/server/server.ts

```diff
@@ -75,7 +75,8 @@ io.on('connection', (socket) => {
 
     //Add Player Name
     socket.on("add-name", (name)=>{
-        name = addName(name,socket.id)
+        isUniquePlayerName(name) ? addName(name,socket.id) : socket.emit("message", "Name already taken, please choose another name ;))");
```

Comment:
this one also has one catch which I found while testing, if we enter the same name, it will give error and display message which is correct, but when we add another name for same socket, the error message still shows.
To get rid of error message , we have to put this line in if-else block and in the if block and make error-message to null so the previous error is gone, use 
if (isUniquePlayerName(name)) {
      addName(name,socket.id);
      socket.emit("error-message", null);
}
and replace the **message** event name with **error-message** to avoid clashing with other message event